### PR TITLE
handle enclave errors

### PR DESCRIPTION
**Pull request by cipepser**

### CODE REVIEW

File: core/types/src/types.rs [hunk 1]
```diff
@@ -23,24 +24,43 @@ pub type RawPubkey = [u8; PUBKEY_SIZE];
...
 pub type RawChallenge = [u8; RANDOMNESS_SIZE];
 
+/// Status for Ecall
 #[repr(C)]
-#[derive(Debug, Clone, Copy, PartialEq)]
-pub enum EnclaveReturn {
-    /// Success, the function returned without any failure.
-    Success,
+#[derive(Debug, PartialEq, Eq)]
+pub struct EnclaveStatus(pub u32);
```
Comment (osuketh) [inline core/types/src/types.rs:30]: * Is it possible to use `enum EnclaveStatus {default, success, error}`?

File: core/enclave/src/bridges/ecalls.rs [hunk 2]
```diff
@@ -65,35 +90,63 @@ pub unsafe extern "C" fn ecall_get_state(
...
-    let pubkey = PublicKey::from_bytes(&pubkey[..])
-        .expect("Failed to read public key.");
-    let key = UserAddress::from_sig(&challenge[..], &sig, &pubkey)
-        .expect("Failed to generate user address.");
+) -> EnclaveStatus {
+    let sig = match Signature::from_bytes(&sig[..]) {
+        Ok(sig) => sig,
+        Err(_) => {
+            debug!("Failed to read signatures.");
+            return EnclaveStatus(1);
```
Comment (osuketh) [inline core/enclave/src/bridges/ecalls.rs:98]: Define `EnclaveStatus::error()` cause `EnclaveStatus(1)` is not easy to understand.

File: core/enclave/src/bridges/ocalls.rs [hunk 3]
```diff
@@ -1,38 +1,38 @@
...
             &mut ias_sock as *mut i32
         )
     };
 
     if status != sgx_status_t::SGX_SUCCESS {
 		return Err(EnclaveError::SgxError{ err: status });
 	}
-	if rt != sgx_status_t::SGX_SUCCESS {
-		return Err(EnclaveError::SgxError{ err: rt });
+	if rt != UntrustedStatus::success() {
```
Comment (osuketh) [inline core/enclave/src/bridges/ocalls.rs:21]: I think had better to define `fn is_success(&self) -> bool` to `UntrustedStatus` and then call `rt.is_success()` here.

Comment (cipepser) [reply core/enclave/src/bridges/ocalls.rs:21]: You mean calling `is_err()` ?
I agree with it is more  easy to understand, so I'll fix it (and `EnclaveStatus` in ecall as well).

**Pull request merged**